### PR TITLE
Feature-gated serde impls for Triple

### PR DESCRIPTION
**Pull request by akesson**
I hope you'll accept this contribution.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,8 +11,15 @@ license = "Apache-2.0 WITH LLVM-exception"
 repository = "https://github.com/bytecodealliance/target-lexicon"
 edition = "2018"
 
+[dependencies]
+serde = { version = "1.0", features = ["derive"], optional = true }
```
Comment (bjorn3) [inline Cargo.toml:15]: The derive feature is not used, right?

File: Cargo.toml [hunk 2]
```diff
@@ -11,8 +11,15 @@ license = "Apache-2.0 WITH LLVM-exception"
...
+
+[dev-dependencies]
+serde_json = "1.0"
+
 [features]
 default = []
+serde = ["dep:serde", "std"]
```
Comment (bjorn3) [inline Cargo.toml:22]: Looks like the MSRV cargo doesn't support this syntax.

File: Cargo.toml (hunk 1 above)
Comment (akesson) [reply Cargo.toml:15]: Nope, sloppy of me

**Comment by sunfishcode** [comment]
Looks reasonable, and thanks @bjorn3 for reviewing!

**Pull request merged**